### PR TITLE
Sentencepiece custom extension integration

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: Dockerfile.ubuntu [hunk 1]
```diff
@@ -60,7 +60,6 @@ tar xvf boost_1_69_0.tar.gz && cd boost_1_69_0 && ./bootstrap.sh && \
 install
 
 COPY third_party /ovms/third_party/
-
```
Comment (dkalinowski) [inline Dockerfile.ubuntu:63]: ?

File: docs/build_from_source.md [hunk 2]
```diff
@@ -115,6 +115,22 @@ make docker_build NVIDIA=1 OV_USE_BINARY=0 OV_SOURCE_BRANCH=releases/2022/3 OV_C
...
+
+Hint: use together with `OV_USE_BINARY=0` to force building OpenVINO from source. Use `OV_SOURCE_BRANCH` parameter to specify which branch from [OpenVINO repository](https://github.com/openvinotoolkit/openvino) should be used.
+Use together with `OV_CONTRIB_BRANCH` to specify which branch from [OpenVINO contrib](https://github.com/openvinotoolkit/openvino_contrib) repository should be used for Custom Operation extension.
+
+Example:
+```bash
+make docker_build SENTENCEPIECE=1 OV_USE_BINARY=0 OV_SOURCE_BRANCH=releases/2022/3 OV_CONTRIB_BRANCH=releases/2022/3
+```
+```bash
+docker run -it --gpus all -p 9178:9178 -v ${PWD}/models/public/resnet-50-tf:/opt/model openvino/model_server:latest --model_path /opt/model --model_name resnet  --cpu_extension /ovms/lib/libuser_ov_extensions.so
```
Comment (dkalinowski) [inline docs/build_from_source.md:133]: gpus all for sentencepiece is not required

File: Dockerfile.ubuntu [hunk 3]
```diff
@@ -126,6 +126,7 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
 
 ARG JOBS
 ARG ov_use_binary=1
+ARG sentencepiece=0
```
Comment (dtrawins) [inline Dockerfile.ubuntu:129]: ARG ov_source_org=openvinotoolkit

File: Dockerfile.ubuntu [hunk 4]
```diff
@@ -142,9 +143,12 @@ ENV TEST_LOG="/root/.cache/bazel/_bazel_root/bc57d4817a53cab8c785464da57d1983/ex
 
 ################### BUILD OPENVINO FROM SOURCE - buildarg ov_use_binary=0  ############################
 # Build OpenVINO and nGraph (OV dependency) with D_GLIBCXX_USE_CXX11_ABI=0 or 1
-RUN if [ "$ov_use_binary" == "0" ] ; then true ; else exit 0 ; fi ; git clone https://github.com/openvinotoolkit/openvino --recurse-submodules --branch $ov_source_branch --depth 1 /openvino
+RUN if [ "$sentencepiece" == "1" ] ; then true ; else exit 0 ; fi ; git clone https://github.com/openvinotoolkit/openvino_contrib.git --recurse-submodules --branch $ov_contrib_branch --depth 1 /openvino_contrib
```
Comment (dtrawins) [inline Dockerfile.ubuntu:146]: Clone if sencencepice=1 or nvidia=1 from ov_source_org

File: Dockerfile.ubuntu [hunk 5]
```diff
@@ -142,9 +143,12 @@ ENV TEST_LOG="/root/.cache/bazel/_bazel_root/bc57d4817a53cab8c785464da57d1983/ex
...
+RUN if [ "$sentencepiece" == "1" ] ; then true ; else exit 0 ; fi ; git clone https://github.com/slyalin/openvino.git --recurse-submodules --branch muse_leftovers --depth 1 /openvino
+RUN if [ "$ov_use_binary" == "0" ] ; then true ; else exit 0 ; fi ; if [ "$sentencepiece" == "0" ] ; then true ; else exit 0 ; fi ; git clone https://github.com/openvinotoolkit/openvino --recurse-submodules --branch $ov_source_branch --depth 1 /openvino
 WORKDIR /openvino/build
-RUN if [ "$ov_use_binary" == "0" ] ; then true ; else exit 0 ; fi ; cmake -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE -DENABLE_SAMPLES=0 -DNGRAPH_USE_CXX_ABI=1 -DCMAKE_CXX_FLAGS=" -D_GLIBCXX_USE_CXX11_ABI=1 -Wno-error=parentheses "  ..
+RUN if [ "$sentencepiece" == "1" ] ; then true ; else exit 0 ; fi ; cmake -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE -DOPENVINO_EXTRA_MODULES=/openvino_contrib/modules -DBUILD_nvidia_plugin=OFF -DBUILD_arm_plugin=OFF -DBUILD_mo_pytorch=OFF -DBUILD_optimum=OFF -DENABLE_SAMPLES=0 -DNGRAPH_USE_CXX_ABI=1 -DCMAKE_CXX_FLAGS=" -D_GLIBCXX_USE_CXX11_ABI=1 -Wno-error=parentheses "  ..
```
Comment (dtrawins) [inline Dockerfile.ubuntu:150]: it is building all cpu extensions, should we filer it or change the arg name to custom_ops ?

**Review by dtrawins — APPROVED** [review]

**Pull request merged**